Userland/Libraries/LibPthread/semaphore.cpp

```diff
@@ -81,16 +92,23 @@ int sem_post(sem_t* sem)
 
     sem->value++;
 
-    pthread_cond_signal(&sem->cv);
+    errno = pthread_cond_signal(&sem->cv);
```

Comment:
Not an issue with your impl, but these `pthread_cond_*` functions will blow up if something goes wrong, so control won't reach here anyway.
Also, they don't seem to return `errno` anyway, `futex()` sets `errno` and `pthread_cond_signal` just blows up if `ret < 0`.

Reply:
I'm OK with things blowing up when `sem->cv` isn't a valid condvar or if something else bad happens. As for `errno`, I know that `pthread_*` doesn't set errno and just returns the error number instead. However, the `sem_post` caller expects the error code in `errno` so that's why I'm copying it to `errno` for them.

Reply:
it doesn't look like `pthread_cond_signal()` (for example) is returning the error.
It asserts that the return value of `futext()` is not -1 (`futex()` returns -1 and sets `errno` if something goes wrong), and then just returns zero.

see `pthread.cpp:519` (for `pthread_cond_signal()`), and `serenity.cpp:97` (for `futex()`)

Reply:
Aye, you're right - that's what it does at the moment. But, according to POSIX.1 it _could_ return an error code at some later point, like `EINVAL` if someone decides to implement that (not that I see why anyone would want to - for performance reasons). So I guess my code is overly cautious.